src/Tester/TestStreamProviders/Generator/GeneratorPooledCache.cs

```diff
@@ -0,0 +1,197 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Orleans.Providers.Streams.Common;
+using Orleans.Serialization;
+using Orleans.Streams;
+using Tester.TestStreamProviders.Generator.Generators;
+
+namespace Tester.TestStreamProviders.Generator
+{
+    public class GeneratorPooledCache : IQueueCache
+    {
+        private readonly IObjectPool<FixedSizeBuffer> bufferPool;
+        private readonly ICacheDataAdapter<GeneratedBatchContainer, CachedMessage> operations;
+        private readonly PooledQueueCache<GeneratedBatchContainer, CachedMessage> cache;
+
+        public GeneratorPooledCache(IObjectPool<FixedSizeBuffer> bufferPool)
+        {
+            if (bufferPool == null)
+            {
+                throw new ArgumentNullException("bufferPool");
+            }
+            this.bufferPool = bufferPool;
+            operations = new CacheDataAdapter(bufferPool, disposable => cache.Purge(disposable));
+            cache = new PooledQueueCache<GeneratedBatchContainer, CachedMessage>(operations);
+        }
```

Comment:
albeit safe, I'd initialize the `cache` field before the `operations` field, as it uses the cache object in its closure. FxCop would even complain that you are passing a reference to `this` before the constructor finished, so even more correct would be to initialize a `queueCache` variable and use that in the closure (as well as assign that same value to the `cache` field).